File: easy/python/0066.py
class Solution:
    def plusOne(self, digits: 'List[int]') -> 'List[int]':
        index = len(digits) - 1
        while(index != -1):
            if digits[index] == 9: 
                digits[index] = 0
            else:
                digits[index] += 1
                return digits
            index -= 1
        return [1] + digits
        
        

File: easy/c/0066.c
int* plusOne(int* digits, int digitsSize, int* returnSize){
    int head = digitsSize - 1;
    while(head > -1){
        if(digits[head] == 9)digits[head] = 0;
        else{
            digits[head] += 1;
            *returnSize = digitsSize;
            return digits;
        }
        head--;
    }
    int* res = calloc((digitsSize + 1), sizeof(int));
    if(digits[0] == 10)for(int i = 1 ; i <= digits; i++) res[i] = digits[i-1];
    res[0] = 1;
    *returnSize = digitsSize + 1;
    return res;
}